src/java/org/apache/cassandra/service/opstate/KeyspaceColumnFamilyOpStateCache.java

```diff
@@ -34,56 +33,63 @@
 {
-    private final ConcurrentMap<KeyspaceTableKey, Instant> tableEntries;
+    private final ConcurrentMap<KeyspaceColumnFamilyKey, Instant> cfEntries;
 
-    public KeyspaceTableOpStateCache(Map<KeyspaceTableKey, Instant> initialEntries)
+    public KeyspaceColumnFamilyOpStateCache(Map<KeyspaceColumnFamilyKey, Instant> initialEntries)
     {
-        this.tableEntries = new ConcurrentHashMap<>(initialEntries);
+        this.cfEntries = new ConcurrentHashMap<>(initialEntries);
     }
 
     @VisibleForTesting
-    Map<KeyspaceTableKey, Instant> getTableEntries()
+    Map<KeyspaceColumnFamilyKey, Instant> getCfEntries()
     {
-        return Collections.unmodifiableMap(tableEntries);
+        return Collections.unmodifiableMap(cfEntries);
     }
 
-    public boolean entryExists(KeyspaceTableKey entryKey)
+    public boolean entryExists(KeyspaceColumnFamilyKey entryKey)
     {
-        return tableEntries.containsKey(entryKey);
+        return cfEntries.containsKey(entryKey);
     }
 
-    public Map<KeyspaceTableKey, Instant> updateTsForEntry(KeyspaceTableKey entryKey, Instant value)
+    public Map<KeyspaceColumnFamilyKey, Instant> updateTsForEntry(KeyspaceColumnFamilyKey entryKey, Instant value)
         throws IllegalArgumentException
     {
-        if (tableEntries.containsKey(entryKey) && tableEntries.get(entryKey).compareTo(value) > 0)
+        if (cfEntries.containsKey(entryKey) && cfEntries.get(entryKey).compareTo(value) > 0)
             throw new IllegalArgumentException("Can only update cache entry with increasing timestamp");
 
-        tableEntries.put(entryKey, value);
-        return Collections.unmodifiableMap(tableEntries);
+        cfEntries.put(entryKey, value);
+        return Collections.unmodifiableMap(cfEntries);
     }
 
     public Optional<Instant> getMinimumTsOfAllEntries()
     {
-        if (tableEntries.isEmpty())
+        if (cfEntries.isEmpty())
             return Optional.empty();
 
         // Remove potentially deleted keyspaces and tables from entries
-        Set<KeyspaceTableKey> invalidEntries =
-            Sets.difference(tableEntries.keySet(), Sets.intersection(tableEntries.keySet(), getValidKeyspaceTableEntries()));
-        invalidEntries.forEach(tableEntries::remove);
-
-        return Optional.of(Collections.min(tableEntries.values()));
+        Set<KeyspaceColumnFamilyKey> validEntries = getValidKeyspaceColumnFamilyEntries();
+        Set<KeyspaceColumnFamilyKey> invalidEntries =
+            Sets.difference(cfEntries.keySet(), Sets.intersection(cfEntries.keySet(), validEntries));
+        invalidEntries.forEach(cfEntries::remove);
+
+        // If not all keyspaces/tables have a tracked last cleanup ts, default to empty value
+        Set<KeyspaceColumnFamilyKey> missingEntries = Sets.difference(validEntries, cfEntries.keySet());
+        if (!missingEntries.isEmpty())
+        {
+            return Optional.empty();
+        }
+        return Optional.of(Collections.min(cfEntries.values()));
```

Comment:
Pref ternary op here and also reverse condition to return `Optional.of(Collections.min(cfEntries.values()))` if  `missingEntries` is empty to simplify reading this:

```java
return missingEntries.isEmpty() ? Optional.of(Collections.min(cfEntries.values())) : Optional.empty();
```